phpmyfaq/src/phpMyFAQ/Controller/Administration/NewsController.php

```diff
@@ -0,0 +1,178 @@
+use phpMyFAQ\Filter;
+use phpMyFAQ\News;
+use phpMyFAQ\Session\Token;
+use phpMyFAQ\Translation;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class NewsController extends AbstractController
+{
+    #[Route('admin/api/news/add')]
+    public function addNews(Request $request)
+    {
+        $this->userHasPermission(PermissionType::NEWS_ADD);
+        $data = json_decode($request->getContent());
+
+        $faqConfig = Configuration::getConfigurationInstance();
+        $news = new News($faqConfig);
+
+        if (!Token::getInstance()->verifyToken('save-news', $data->csrfToken)) {
+            return $this->json(['error' => Translation::get('err_NotAuth')], Response::HTTP_UNAUTHORIZED);
+        }
+
+        $dateStart = Filter::filterVar($data->dateStart, FILTER_SANITIZE_SPECIAL_CHARS);
+        $dateEnd = Filter::filterVar($data->dateEnd, FILTER_SANITIZE_SPECIAL_CHARS);
+        $header = Filter::filterVar($data->newsHeader, FILTER_SANITIZE_SPECIAL_CHARS);
+        $content = Filter::filterVar($data->news, FILTER_SANITIZE_SPECIAL_CHARS);
+        $author = Filter::filterVar($data->authorName, FILTER_SANITIZE_SPECIAL_CHARS);
+        $email = Filter::filterVar($data->authorEmail, FILTER_VALIDATE_EMAIL);
+        $active = Filter::filterVar($data->active, FILTER_SANITIZE_SPECIAL_CHARS);
+        $comment = Filter::filterVar($data->comment, FILTER_SANITIZE_SPECIAL_CHARS);
+        $link = Filter::filterVar($data->link, FILTER_SANITIZE_SPECIAL_CHARS);
+        $linkTitle = Filter::filterVar($data->linkTitle, FILTER_SANITIZE_SPECIAL_CHARS);
+        $newsLang = Filter::filterVar($data->langTo, FILTER_SANITIZE_SPECIAL_CHARS);
+        $target = Filter::filterVar($data->target, FILTER_SANITIZE_SPECIAL_CHARS);
+
+        $newsMessage = new NewsMessage();
+        $newsMessage
+            ->setLanguage($newsLang)
+            ->setHeader($header)
+            ->setMessage(html_entity_decode((string)$content))
+            ->setAuthor($author)
+            ->setEmail($email)
+            ->setActive($active)
+            ->setComment($comment)
+            ->setDateStart(new DateTime($dateStart))
+            ->setDateEnd(new DateTime($dateEnd))
+            ->setLink($link ?? '')
+            ->setLinkTitle($linkTitle ?? '')
+            ->setLinkTarget($target ?? '')
+            ->setCreated(new DateTime());
+
+        if ($news->create($newsMessage)) {
+            return $this->json(['success' => Translation::get('ad_news_updatesuc')], Response::HTTP_OK);
+        } else {
+            return $this->json(['error' => Translation::get('err_NotAuth')], Response::HTTP_UNAUTHORIZED);
+        }
+    }
+
+    #[Route('admin/api/news/delete')]
+    public function deleteNews(Request $request)
+    {
+        $this->userHasPermission(PermissionType::NEWS_DELETE);
+        $data = json_decode($request->getContent());
+
+        $faqConfig = Configuration::getConfigurationInstance();
```

Comment:
You don't need this, you can get the configuration instance from

`$this->configuration`

as it's defined and instantiated in the AbstractController